torchlight.js

```diff
@@ -1,1047 +1,1072 @@
+									break;
+								case "Type4":
+									updateTokenLighting(5,5, "#a2642a", "0.7", 360, "torch", 10, 7);
+									break;
+								case "Type5":
+									updateTokenLighting(5,5, "#a2642a", "0.5", 360, "torch", 10, 5);
+									break;
+								case "Type6":
+									updateTokenLighting(5,5, "#a2642a", "0.3", 360, "torch", 10, 3);
+									break;
+								case "Type7":
+									updateTokenLighting(nBright*2,nDim*2, "#a2642a", "0.7", 60, "torch", 10, 7);
+									break;
+								case "Type8":
+									updateTokenLighting(nBright*2,nDim*2, "#a2642a", "0.5", 60, "torch", 10, 5);
+									break;
+								case "Type9":
+									updateTokenLighting(nBright*2,nDim*2, "#a2642a", "0.3", 60, "torch", 10, 3);
+									break;
+								case "TypeC":
+									updateTokenLighting(nBright,nDim,
+										game.settings.get("torchlight", "customLanternColor"),
+										game.settings.get("torchlight", "customLanternColorIntensity"),
+										360,
+										game.settings.get("torchlight", "customLanternAnimationType"),
+										game.settings.get("torchlight", "customLanternAnimationSpeed"),
+										game.settings.get("torchlight", "customLanternAnimationIntensity"));
+									break;
+							}
+						} else {
+							// There is no oil to consume, signal and disable the button
+							ChatMessage.create({
+								user: game.user._id,
+								speaker: game.actors.get(data.actorId),
+								content: "No "+game.settings.get("torchlight", "nameConsumableLantern")+" in Inventory !"
+							});
+							disableTorchlightButton(tbuttonLantern);
+						}
+					}
+				// Or are we dealing with the Torch Button
+				} else if (tbutton === tbuttonTorch) {
+					// Check if the token has the torch on
+					if (statusTorch) {
+						// The token has the torch on
+						console.log("Clicked on the torch button when the torch is on.");
+						statusTorch = false;
+						await app.object.setFlag("torchlight", "statusTorch", false);
+						tbuttonTorch.removeClass("active");
+						// Torch is inactive, enable the relevant light sources according to parameters
+						await enableRelevantButtons();
+						// Restore the initial light source
+						updateTokenLighting(
+							app.object.getFlag("torchlight", "InitialBrightRadius"),
+							app.object.getFlag("torchlight", "InitialDimRadius"),
+							app.object.getFlag("torchlight", "InitialLightColor"),
+							app.object.getFlag("torchlight", "InitialColorIntensity"),
+							app.object.getFlag("torchlight", "InitialLightAngle"),
+							app.object.getFlag("torchlight", "InitialAnimationType"),
+							app.object.getFlag("torchlight", "InitialAnimationSpeed"),
+							app.object.getFlag("torchlight", "InitialAnimationIntensity"),
+						);
+					} else {
+						// The token does not have the torch on
+						console.log("Clicked on the torch when the torch is off.");
+						// Checks whether the character can consume a torch
+						if (consumeItem("Torch")) {

```

Comment:
replace this line with 

if (consumeItem(game.settings.get("torchlight", "nameConsumableTorch"))) {

Reply:
Yes, I missed that.. I already did on my master branch. I am doing some changes to show messages when turning on or off a light/torch/lantern. but all changes are In my repo. I will close this pull request and will do a new one when I finish my changes.